interactions-transitions-assignment/part-2-map/us-station-interaction.js

```diff
@@ -22,15 +22,169 @@ function createMap(elementId) {
         .attr('class', 'svg-group')
         .attr('transform', 'translate(' + margins.left + ',' + margins.top + ')');
 
+    let geoJson;
+    let stations;
+    let nonNullStations;
+    let elevScale;
+    let stationColorKey;
+
     // read in data
-    d3.json('us-states.json').then(function (geoJSONStates) {
-        d3.csv('NSRDB_StationsMeta.csv').then(function (stations) {
-            draw(geoJSONStates, stations);
-        });
+    Promise.all([
+        d3.json('us-states.json'),
+        d3.csv('NSRDB_StationsMeta.csv', d3.autoType)
+    ]).then(function ([geoJSONData, stationData]) {
+        geoJson = geoJSONData;
+        stations = stationData;
+        draw();
     });
```

Comment:
this is well done! I do think that you could potentially seperate the draw function into a base layers function and then also a content layer. This might allow you to just utilize one function for the content layer and reuse it.

Reply:
oh ok i see that's kinda what you've done. looks pretty good!